mirdata/datasets/mdb_stem_synth.py

```diff
@@ -0,0 +1,169 @@
+
+"""
+
+from typing import BinaryIO, Optional, TextIO, Tuple
+
+import librosa
+import numpy as np
+
+import os
+import glob
+
+from mirdata import io, jams_utils, core, annotations, download_utils
+
+BIBTEX = """
+@inproceedings{salamon2017analysis,
+  title={An analysis/synthesis framework for automatic f0 annotation of multitrack datasets},
+  author={Salamon, Justin and Bittner, Rachel M and Bonada, Jordi and Bosch, Juan J and G{\'o}mez Guti{\'e}rrez, Emilia and Bello, Juan Pablo},
+  booktitle={International Society for Music Information Retrieval Conference},
+  year={2017},
+}
+"""
+
+INDEXES = {
+    "default": "1.0.0",
+    "test": "1.0.0",
+    "1.0.0": core.Index(filename="mdb_stem_synth_index_1.0.0.json"),
+}
+
+REMOTES = {
+    "mdb_stem_synth": download_utils.RemoteFileMetadata(
+        filename="MDB-stem-synth.tar.gz",
+        url="https://zenodo.org/records/1481172/files/MDB-stem-synth.tar.gz?download=1",
+        checksum="31c1f6b4888e5fd108af91c69789a809",
+        unpack_directories=["MDB-stem-synth"],
+    )
+}
+
+
+LICENSE_INFO = """
+Attribution-NonCommercial 4.0 International
+"""
+
+
+class Track(core.Track):
+    """mdb_stem_synth Track class
+
+    Args:
+        track_id (str): track id of the track
+
+    Attributes:
+        audio_path (str): path to the track's audio file
+        f0_path (str): path to the track's f0 annotation file
+        track_id (str): track id
+        audio (Tuple[np.ndarray, float]): the track's audio signal and sample rate
+
+    Cached Properties:
+        f0 (F0Data): the track's f0 annotation
+
+    """
+
+    def __init__(self, track_id, data_home, dataset_name, index, metadata):
+        super().__init__(
+            track_id,
+            data_home,
+            dataset_name=dataset_name,
+            index=index,
+            metadata=metadata,
+        )
+
+        self.audio_path = self.get_path("audio")
+        self.f0_path = self.get_path("f0")
+
+    @core.cached_property
+    def f0(self) -> Optional[annotations.F0Data]:
+        return load_f0(self.f0_path)
+
+    @property
+    def audio(self) -> Optional[Tuple[np.ndarray, float]]:
+        """The track's audio
+
+        Returns:
+            * np.ndarray - audio signal
+            * float - sample rate
+
+        """
+        return load_audio(self.audio_path)
+
+    def to_jams(self):
+        """Jams: the track's data in jams format"""
+        return jams_utils.jams_converter(
+            audio_path=self.audio_path,
+            f0_data=[(self.f0, None)],
+        )
+
+
+@io.coerce_to_bytes_io
+def load_audio(fhandle: BinaryIO) -> Tuple[np.ndarray, float]:
+    """Load MDB-stem-synth audio
+
+    Args:
+        fhandle (str or file-like): File-like object or path to audio file
+
+    Returns:
+        * np.ndarray - audio signal
+        * float - sample rate
+
+    """
+    return librosa.load(fhandle, sr=None, mono=True)
+
+
+@io.coerce_to_string_io
+def load_f0(fhandle: str | TextIO) -> annotations.F0Data:
```

Comment:
I believe you'd need to remove the `|` symbol here and stick to `TextIO` only.

Reply:
Ah right, thanks for that. I only tested it in Python 3.11, removed this now